File: src/com/javarush/test/level04/lesson16/home02/Solution.java
package com.javarush.test.level04.lesson16.home02;

import java.io.BufferedReader;
import java.io.InputStreamReader;

/* Среднее такое среднее
Ввести с клавиатуры три числа, вывести на экран среднее из них. Т.е. не самое большое и не самое маленькое.
*/

public class Solution
{
    public static void main(String[] args)   throws Exception
    {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        String sa = reader.readLine();
        String sb = reader.readLine();
        String sc = reader.readLine();
        int a = Integer.parseInt(sa);
        int b = Integer.parseInt(sb);
        int c = Integer.parseInt(sc);
        if (a>b&&b>c||a<b&&b<c) {
            System.out.println(b);
        } else if (a>c&&c>b||a<c&&c<b) {
            System.out.println(c);
        } else if (b>a&&a>c||b<a&&a<c) {
            System.out.println(a);
        }
    }
}

File: src/com/javarush/test/level04/lesson16/home02/Solution.kt
package com.javarush.test.level04.lesson16.home02

import java.io.BufferedReader
import java.io.InputStreamReader

@Throws(Exception::class)
fun main(args: Array<String>) {
    val reader = BufferedReader(InputStreamReader(System.`in`))
    val sa = reader.readLine()
    val sb = reader.readLine()
    val sc = reader.readLine()
    reader.close()
    val a = Integer.parseInt(sa)
    val b = Integer.parseInt(sb)
    val c = Integer.parseInt(sc)

    when {
        (b in (c + 1)..(a - 1) || b in (a + 1)..(c - 1)) -> println(b)
        (c in (b + 1)..(a - 1) || c in (a + 1)..(b - 1)) -> println(c)
        (a in (c + 1)..(b - 1) || a in (b + 1)..(c - 1)) -> println(a)
    }
}